### PR TITLE
Reimplement pagination query for MySQL datastore

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

The previous implementation of MySQL pagination condition ( `UpdatedAt <= ? AND Id > UUID_TO_BIN(?,true)` ) leads to unexpected behavior: Instead of get `UpdatedAt <= ?` first then apply `Id > UUID_TO_BIN(?,true)` in the result set, MySQL makes intersection of 2 sets `UpdatedAt <= ?` and `Id > UUID_TO_BIN(?,true)`. This implementation resolve this issue by using an approximated variant.

ref: https://use-the-index-luke.com/sql/partial-results/fetch-next-page#sb-row-values

**Which issue(s) this PR fixes**:

Related to #1555 
Follow PR #1873 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/datastore/mysql/query.go [hunk 1]
```diff
@@ -87,37 +87,52 @@ func buildPaginationCondition(opts datastore.ListOptions) string {
...
 		} else {
-			conds[i] = fmt.Sprintf("%s %s ?", o.Field, makePaginationConditionOperator(o))
+			subSetConds[i] = fmt.Sprintf("%s = ?", o.Field)
 		}
 	}
 
 	// If there is no filter, mean pagination condition should be treated as the only where condition.
 	if len(opts.Filters) == 0 {
-		return fmt.Sprintf("WHERE %s", strings.Join(conds[:], " AND "))
+		return fmt.Sprintf("WHERE %s AND NOT (%s)", strings.Join(outerSetConds[:], " AND "), strings.Join(subSetConds[:], " AND "))
```
Comment (nghialv) [inline pkg/datastore/mysql/query.go:117]: nit: `outerSetConds[:]` -> `outerSetConds`

**Review by nghialv — COMMENTED** [review]
Nice catch! Just left a nit, everything else is LGTM.
/lgtm

File: pkg/datastore/mysql/query.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/datastore/mysql/query.go]: Nice catch 👍 Addressed by https://github.com/pipe-cd/pipe/pull/1898/commits/2903c6d2fd8722295c7c83569787a1255110a694 🙏

**Comment by nakabonne** [comment]
way to get.
/approve

**Pull request merged**